rustls/src/crypto/mod.rs

```diff
@@ -626,6 +604,47 @@ pub fn default_fips_provider() -> CryptoProvider {
     aws_lc_rs::default_provider()
 }
 
+mod static_default {
+    #[cfg(not(feature = "std"))]
+    use alloc::boxed::Box;
+    use alloc::sync::Arc;
+
+    #[cfg(not(feature = "std"))]
+    use once_cell::race::OnceBox;
+    #[cfg(feature = "std")]
+    use once_cell::sync::OnceCell;
+
+    use super::CryptoProvider;
+
+    #[cfg(feature = "std")]
+    pub(crate) fn install_default(
+        default_provider: CryptoProvider,
+    ) -> Result<(), Arc<CryptoProvider>> {
+        PROCESS_DEFAULT_PROVIDER.set(Arc::new(default_provider))
+    }
+
+    #[cfg(not(feature = "std"))]
+    pub(crate) fn install_default(
+        default_provider: CryptoProvider,
+    ) -> Result<(), Arc<CryptoProvider>> {
```

Comment:
Is changing this return type from `Result<(), Box<Arc<Self>>>` a breaking change for no-std users? It feels like yes.

Reply:
I think so, but it fixes the feature-controlled breaking change which I would argue is kind of worse.

Reply:
True, and I think the no-std userbase is very small at this point.